packages/docs/stories/templates/appshell.stories.tsx

```diff
@@ -0,0 +1,503 @@
+
+      // Update the current indicator
+      nav.querySelectorAll('syn-nav-item').forEach(item => {
+        item.removeAttribute('current');
+        if (item === target) {
+          item.setAttribute('current', '');
+        }
+      });      
+    };
+
+    // Add the listener to the given navigation
+    nav.addEventListener('click', navItemClickEvent);
+  </script>
+`;
+
+const meta: Meta = {
+  parameters: {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    chromatic: {
+      ...storyBookPreviewConfig?.parameters?.chromatic,
+      disableSnapshot: false,
+    },
+    design: generateFigmaPluginObject('8462-8334'),
+    docs: {
+      description: {
+        component: generateStoryDescription('templates', 'application-shell'),
+      },
+      page: () => (
+        <>
+          <Title />
+          <Subtitle />
+          <Description />
+          <Stories title="" />
+        </>
+      ),
+      story: {
+        iframeHeight: 550,
+      },
+    },
+  },
+  title: 'Templates/AppShell',
+};
+export default meta;
+
+export const SideNavigation: Story = {
+  render: () => html`
+    <!-- .synergy-demo-application -->
+    <div class="synergy-demo-application" id="appshell-side-navigation">
+      ${createHeader()}
+      
+      <!-- .synergy-demo-content -->
+      <div class="synergy-demo-content">
+        ${createSideNav()}
+        ${createMainContent()}
+        ${createFooter()}
+      </div>
+      <!-- /.synergy-demo-content -->
+    </div>
+    <!-- /.synergy-demo-application -->
+    ${createSharedStyles()}
+    ${createSidebarConnector('appshell-side-navigation')}
+    ${createDemoNavigation('appshell-side-navigation')}
+  `,
+};
+
+export const SideNavigationShrinkingContent: Story = {
+  render: () => html`
+    <!-- .synergy-demo-application -->
+    <div class="synergy-demo-application" id="appshell-shrink">
+      ${createHeader()}
+      
+      <!-- .synergy-demo-content -->
+      <div class="synergy-demo-content">
+        ${createSideNav({ open: true })}
+        <div class="synergy-demo-content-inner">
+          ${createMainContent()}
+          ${createFooter()}
+        </div>
+      </div>
+      <!-- /.synergy-demo-content -->
+
+    </div>
+    <!-- /.synergy-demo-application -->
+    ${createSharedStyles()}
+    <style>
+      #appshell-shrink {
+        /**
+         * Set this variable to the wanted size of the side-nav
+         * Make sure the value stays in sync, otherwise the parts my overlap
+         */
+        --appshell-shrink-nav-open-width: 25rem;
+
+        syn-side-nav::part(overlay) {
+          display: none;
+        }
+
+        .synergy-demo-content-inner {
+          /**
+           * Set the initial margin left. Needed because we
+           * create the page with the side-bar open
+           */
+          margin-left: var(--appshell-shrink-nav-open-width);
+          transition: margin-left 250ms;
```

Comment:
This should only happen if reduced motion is not active. Please make a query for that.

Reply:
I just had a look as I was not able to find a call to `prefers-reduced-motion` in our other animations. It seems that shoelace handles this with its `animate` js function, but not when using inline transforms or transitions in css (e.g. in checkbox there are fixed transitions without `prefers-reduced-motion`, too).

I would add it here, but we also have to think about the other uses of inline transitions.

Reply:
It's about _reduced_ motion, not removing _every_ motion.

Currently the drawer doesn't have a motion effect, when you activate "Reduce motion" in the settings:

![CleanShot 2024-05-16 at 09 50 09](https://github.com/synergy-design-system/synergy-design-system/assets/26542182/508a9d42-e95c-4cdf-8a2a-f360c3978d38)

The problem here is: The drawer skips the transition, while the content always does the transition, which doesn't fit together here. :) So we have to adapt the template, that it works together from a11y perspective.

Reply:
Thats exactly what I meant. I did a quick search and saw that shoelace does this in its own animate function, but not in its custom css transitions, which we will also have to check as our own components may also animate (which may be unwanted).

Reply:
Done in latest